spec.emu

```diff
@@ -1545,6 +1545,37 @@ return lastURL;</code></pre>
   </emu-clause>
 </emu-clause>
 
+<emu-clause id="sec-operations-on-source-map-records">
+  <h1>Operations on source map records</h1>
+
+  <p>After decoding a source map, source map consumers can use the resulting Decoded Source Map Records to look up position information for debugging or other use cases. This section describes the behaviour of typical operations that may be supported by source map consumers.</p>
+  <p>The GetOriginalPositions operation may be used to query the positions in an original source that correspond to a position in the generated code. For example, this can be used in a debugger to navigate from the generated code to the original source based on a user's mouse click.</p>
+
+  <emu-clause id="sec-GetOriginalPositions" type="abstract operation">
+    <h1>
+      GetOriginalPositions (
+        _sourceMapRecord_: a Decoded Source Map Record,
+        _generatedPosition_: a Position Record,
+      ): a List of Original Position Records
+    </h1>
+    <dl class="header"></dl>
+    <emu-alg>
+      1. Let _mappings_ be _sourceMapRecord_.[[Mappings]].
+      1. Let _last_ be *null*.
+      1. Let _originalPositions_ be a new empty List.
+      1. For each element _mapping_ of _mappings_, in reverse List order, do
```

Comment:
Nit: why not do a forward scan, instead of a reverse _and_ forward scan?

```js
let matches = [];
for (const mapping of mappings) {
  const compare = comparePos(mapping, needle);
  if (compare === EQUAL) {
    matches.push(mapping);
  } else if (compare === GREATER) {
    break;
  }
}
```

Reply:
As it's currently written, the function returns all the mappings at the equal generated position if there are any, or all the mappings that's at the greatest lower bound generated position if there aren't equal mappings.

If you only do a forward scan you find the equal mappings easily but you have to do more work for the lower bound.